File: addons_13/addons_custom/odoo-custom-addons/resp/survey_score_template/models/survey_employee.py
# _*_ coding:utf_8 _*_
from odoo import models, fields, api
from uuid import uuid4

class SurveyEmployee(models.Model):
    _name = 'survey_score_template.survey_employee'
    _rec_name = 'user_id'

    user_id = fields.Many2one('res.users',string='user_id')
    job_id = fields.Many2one('hr.job',string='job_id')
    department_id = fields.Many2one('hr.department',string='department_id')
    sex = fields.Char(string='sex')
    year = fields.Char(string='year')
    level_sf = fields.Char(string='level_sf')
    level_primary = fields.Char(string='level_primary')
    level_secondary = fields.Char(string='level_secondary')
    level_preparatory = fields.Char(string='level_preparatory')
    level_technical = fields.Char(string='level_technical')
    level_degree = fields.Char(string='level_degree')
    level_mastery = fields.Char(string='level_mastery')
    level_doctorate = fields.Char(string='level_doctorate')
    job_type = fields.Char(string='job_type')
    contract_type = fields.Char(string='contract_type')
    personal_type = fields.Char(string='personal_type')
    jornal_type = fields.Char(string='jornal_type')
    turn = fields.Char(string='turn')
    experience_job = fields.Char(string='experience_job')
    experience_work = fields.Char(string='experience_work')
    token = fields.Char(string='token',default=lambda self: self._get_uuid())
    state = fields.Selection(
        selection=[
                ('draft', 'Borrador'),
                ('terms', 'Terminos'),
                ('employee', 'Datos de empleado'),
                ('traumatic', U'Acontecimientos traumáticos'),
                ('in_surveys', 'En encuestas'),
                ('closed', 'Finalizado'),
        ], default='draft')
    survey = fields.Many2one('survey.survey')
    survey_answer = fields.Integer(string='survey_answer',default=0)
    surveys_to_answer = fields.Integer(string='surveys_to_answer',default=0)
    traumatic = fields.Many2one('survey_score_template.survey_traumatic')

    _sql_constraints = [('unique_id','UNIQUE(user_id)',"Error, el usuario ya existe registrado")]

    def _get_uuid(self):
        return str(uuid4())

File: addons_13/addons_custom/odoo-custom-addons/resp/survey_score_template/static/src/js/survey_employee.js
odoo.define('survey_score_template.employee', function (require) {
'use strict';

  require('web.dom_ready');
  var core = require('web.core');
  var _t = core._t;

  var level_sf = $("input[name='level-sf']");
  var level_primary = $("input[name='level-primary']");
  var level_secondary = $("input[name='level-secondary']");
  var level_preparatory = $("input[name='level-preparatory']");
  var level_technical = $("input[name='level-technical']");
  var level_degree = $("input[name='level-degree']");
  var level_mastery = $("input[name='level-mastery']");
  var level_doctorate = $("input[name='level-doctorate']");

  $('#form_employee').on('submit', function(e) {
    var input = $("input[type=hidden]");
    var sex = $("input[name='sex']:checked").length;
    var year = $("input[name='year']:checked").length;
    var job_type = $("input[name='job_type']:checked").length;
    var contract_type = $("input[name='contract_type']:checked").length;
    var personal_type = $("input[name='personal_type']:checked").length;
    var jornal_type = $("input[name='jornal_type']:checked").length;
    var turn = $("input[name='turn']:checked").length;
    var experience_job = $("input[name='experience_job']:checked").length;
    var experience_work = $("input[name='experience_work']:checked").length;
    var nulls = 0;

    if (sex == 0) {
      $(".sex").css({'background-color':'#ffd4d8'});
      nulls = 1;
    }
    else{
      $(".sex").css({'background-color':'#ffffff'});
    }

    if (year == 0) {
      $(".year").css({'background-color':'#ffd4d8'});
      nulls = 1;
    }
    else{
      $(".year").css({'background-color':'#ffffff'});
    }

    if (job_type == 0) {
      $(".job_type").css({'background-color':'#ffd4d8'});
      nulls = 1;
    }
    else{
      $(".job_type").css({'background-color':'#ffffff'});
    }

    if (contract_type == 0) {
      $(".contract_type").css({'background-color':'#ffd4d8'});
      nulls = 1;
    }
    else{
      $(".contract_type").css({'background-color':'#ffffff'});
    }

    if (personal_type == 0) {
      $(".personal_type").css({'background-color':'#ffd4d8'});
      nulls = 1;
    }
    else{
      $(".personal_type").css({'background-color':'#ffffff'});
    }

    if (jornal_type == 0) {
      $(".jornal_type").css({'background-color':'#ffd4d8'});
      nulls = 1;
    }
    else{
      $(".jornal_type").css({'background-color':'#ffffff'});
    }

    if (turn == 0) {
      $(".turn").css({'background-color':'#ffd4d8'});
      nulls = 1;
    }
    else{
      $(".turn").css({'background-color':'#ffffff'});
    }

    if (experience_job == 0) {
      $(".experience_job").css({'background-color':'#ffd4d8'});
      nulls = 1;
    }
    else{
      $(".experience_job").css({'background-color':'#ffffff'});
    }

    if (experience_work == 0) {
      $(".experience_work").css({'background-color':'#ffd4d8'});
      nulls = 1;
    }
    else{
      $(".experience_work").css({'background-color':'#ffffff'});
    }

    if (nulls == 1) {
      e.preventDefault();
    }

  });
  $(level_sf).change(function() {
    if (this.checked) {
      $(level_primary).prop("disabled",true).prop('checked',false);
      $(level_secondary).prop("disabled",true).prop('checked',false);
      $(level_preparatory).prop("disabled",true).prop('checked',false);
      $(level_degree).prop("disabled",true).prop('checked',false);
      $(level_mastery).prop("disabled",true).prop('checked',false);
      $(level_doctorate).prop("disabled",true).prop('checked',false);
      $(level_technical).prop("disabled",true).prop('checked',false);
    }
    else{
      $(level_primary).prop("disabled",false);
      $(level_secondary).prop("disabled",false);
      $(level_preparatory).prop("disabled",false);
      $(level_degree).prop("disabled",false);
      $(level_mastery).prop("disabled",false);
      $(level_doctorate).prop("disabled",false);
      $(level_technical).prop("disabled",false);
    }
  });

});